client/web/src/enterprise/batches/detail/BulkOperationsTab.tsx

```diff
@@ -0,0 +1,73 @@
+import * as H from 'history'
+import MapSearchIcon from 'mdi-react/MapSearchIcon'
+import React, { useCallback } from 'react'
+import { delay, repeatWhen, tap } from 'rxjs/operators'
+
+import { dismissAlert } from '../../../components/DismissibleAlert'
+import { FilteredConnection, FilteredConnectionQueryArguments } from '../../../components/FilteredConnection'
+import { BulkOperationFields, Scalars } from '../../../graphql-operations'
+
+import { queryBulkOperations as _queryBulkOperations } from './backend'
+import { BulkOperationNode, BulkOperationNodeProps } from './bulk-operations/BulkOperationNode'
+
+export interface BulkOperationsTabProps {
+    batchChangeID: Scalars['ID']
+    history: H.History
+    location: H.Location
+
+    queryBulkOperations?: typeof _queryBulkOperations
+}
+
+export const BulkOperationsTab: React.FunctionComponent<BulkOperationsTabProps> = ({
+    batchChangeID,
+    history,
+    location,
+    queryBulkOperations = _queryBulkOperations,
+}) => {
+    const query = useCallback(
+        ({ first, after }: FilteredConnectionQueryArguments) =>
+            queryBulkOperations({ batchChange: batchChangeID, after: after ?? null, first: first ?? null }).pipe(
+                tap(connection => {
+                    for (const node of connection.nodes) {
+                        // Hide alerts for bulk operations seen already.
+                        dismissAlert(`bulkOperation-${node.state.toLocaleLowerCase()}-${node.id}`)
+                    }
+                }),
+                repeatWhen(notifier => notifier.pipe(delay(2000)))
+            ),
+        [batchChangeID, queryBulkOperations]
+    )
+    return (
+        <FilteredConnection<BulkOperationFields, Omit<BulkOperationNodeProps, 'node'>>
+            className="mt-2"
+            nodeComponent={BulkOperationNode}
+            nodeComponentProps={{ showErrors: true }}
+            queryConnection={query}
+            hideSearch={true}
+            defaultFirst={15}
+            noun="bulk operation"
+            pluralNoun="bulk operations"
+            history={history}
+            location={location}
+            useURLQuery={true}
+            listComponent="div"
+            listClassName="mb-3"
+            emptyElement={<EmptyBulkOperationsListElement />}
+            noSummaryIfAllNodesVisible={true}
+            headComponent={BulkOperationsListHeadComponent}
+        />
+    )
+}
+
+export const EmptyBulkOperationsListElement: React.FunctionComponent<{}> = () => (
+    <div className="text-muted mt-4 pt-4 mb-4 row">
+        <div className="col-12 text-center">
```

Comment:
It seems the empty list element isn't exactly centered when it's rendered:

![image](https://user-images.githubusercontent.com/8942601/118552544-eccc3680-b713-11eb-80c7-a09f38aaade6.png)


I suspect the pair of these two lines is causing the conflict -- do we maybe want this?

```suggestion
    <div className="text-center text-muted mt-4 pt-4 mb-4 col-12">
```

Reply:
We found out this was due to @courier-new  having the redesign toggle enabled 😬 Simplified it regardless, thanks for the suggestion :)